IN-CORE/pyincore
# 469 update retrofit dataset

add additional function to update the existing inventory dataset with additional retrofit and mapping entry key information. https://github.com/IN-CORE/pyincore/pull/470/files#diff-d1b1cb98d647ccb4ebd9f558b2894472d298769fdd3c2bde8d9d4e00205e4de0R73

Since we use fiona object and looping through fiona is too slow, I used geopandas dataframe to merge and udpate; but then save to shapefile so we can resume using fiona to carry out normal fragility matching and calculation

----

### To test:
- Use this notebook to test SLC retrofit strategy
[SLC_retrofit_test.zip](https://github.com/IN-CORE/pyincore/files/14240182/SLC_retrofit_test.zip)
- Use this notebook to tests Galveston
[Galveston_retrofit_test.zip](https://github.com/IN-CORE/pyincore/files/14354234/Galveston_retrofit_test.zip)
- RUN `tests/pyincore/analyses/buildingdamage/test_buildingdamage_retrofit.py` and see results in the retrofit folder

Note:
- Joplin takes 31 seconds; 
- Lumberton takes 61 seconds.
- SLC if only retrofit 5000+ building with inner join total runtime: 158 seconds;
- Galveston retrofitting 1500 buildings combined damage takes 115 seconds
